csharp/ql/lib/upgrades/initial/semmlecode.csharp.dbscheme

```diff
@@ -1,1707 +1,752 @@
-/*
- * External artifacts
- */
-
-externalDefects(
-  unique int id: @externalDefect,
-  varchar(900) queryPath: string ref,
-  int location: @location ref,
-  varchar(900) message: string ref,
-  float severity: float ref);
-
-externalMetrics(
-  unique int id: @externalMetric,
-  varchar(900) queryPath: string ref,
-  int location: @location ref,
-  float value: float ref);
-
-externalData(
-  int id: @externalDataElement,
-  varchar(900) path: string ref,
-  int column: int ref,
-  varchar(900) value: string ref);
-
-snapshotDate(
-  unique date snapshotDate: date ref);
-
-sourceLocationPrefix(
-  varchar(900) prefix: string ref);
-
-/*
- * Duplicate code
- */
-
-duplicateCode(
-  unique int id: @duplication,
-  varchar(900) relativePath: string ref,
-  int equivClass: int ref);
-
-similarCode(
-  unique int id: @similarity,
-  varchar(900) relativePath: string ref,
-  int equivClass: int ref);
-
-@duplication_or_similarity = @duplication | @similarity
-
-tokens(
-  int id: @duplication_or_similarity ref,
-  int offset: int ref,
-  int beginLine: int ref,
-  int beginColumn: int ref,
-  int endLine: int ref,
-  int endColumn: int ref);
-
-/*
- * Version history
- */
-
-svnentries(
-  int id: @svnentry,
-  varchar(500) revision: string ref,
-  varchar(500) author: string ref,
-  date revisionDate: date ref,
-  int changeSize: int ref);
-
-svnaffectedfiles(
-  int id: @svnentry ref,
-  int file: @file ref,
-  varchar(500) action: string ref);
-
-svnentrymsg(
-  int id: @svnentry ref,
-  varchar(500) message: string ref
-)
-
-svnchurn(
-  int commit: @svnentry ref,
-  int file: @file ref,
-  int addedLines: int ref,
-  int deletedLines: int ref);
-
-/*
- * C# dbscheme
- */
-
-/** ELEMENTS **/
-
-@element = @declaration | @stmt | @expr | @modifier | @attribute | @namespace_declaration
-         | @using_directive | @type_parameter_constraints | @external_element
-         | @xmllocatable | @asp_element | @namespace;
-
-@declaration = @callable | @generic | @assignable;
```

Comment:
This looks like a significant change to the "initial" dbscheme. 
(1) Does the replacement initial "match" the old dbscheme in one of the upgrade folders? 
(2) Does it require that the hash value is re-calculated for the naming purposes of that particular upgrade folder.
(3) Or was the initial db scheme here just plain wrong 😄 ?

Reply:
I would also like to know what was wrong with the original initial DB scheme for C#?